packages/hub/src/scope.ts

```diff
@@ -465,7 +465,7 @@ export class Scope implements ScopeInterface {
       event.level = this._level;
     }
     if (this._transactionName) {
-      event.transaction = this._transactionName;
+      event.transaction = event.transaction || this._transactionName;
```

Comment:
Can we add a comment here on why we did this? I.e. a one- or two-liner saying that we don't want to set `event.transaction` for transaction events and that it works because transaction events have a `transaction` field set from the get-go.